src/main/java/com/minecolonies/util/StructureInfo.java

```diff
@@ -0,0 +1,90 @@
+package com.minecolonies.util;
+
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * Created by Dars on 9/21/2016.
+ */
+public class StructureInfo
+{
+
+    /**
+     * The name this schematic has.
+     */
+    private String name;
+
+    /**
+     * The category  of the structure.
+     */
+    private String category = "hut";
+
+    private Boolean isHut = false;
+
+    /**
+     * style, level, path
+     */
+    private final Map<String, TreeMap<Integer, Path>> resourceLocationsByStyleAndLevel = new HashMap<>();
+
+    /**
+     *
+     * @param name the name of the structure (hut or decoration)
+     * @param isHut is it a hut- meaning does it have a hut block
+     * @param category the category the builders tool should display it under
+     */
+    public StructureInfo(String name, Boolean isHut, String category)
+    {
+        this.name = name;
+        this.isHut = isHut;
+        this.category=category;
+    }
+
+    public String getName()
+    {
+        return this.name;
+    }
+
+    public String getCategory()
+    {
+        return this.category;
+    }
+
+    public int getMaxLevel(String style)
+    {
+        if(!this.resourceLocationsByStyleAndLevel.containsKey(style))
+        {
+            return 0;
+        }
+        Integer maxLevel =
+          Collections.max(this.resourceLocationsByStyleAndLevel.get(style).keySet());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "maxLevel". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1488)